packages/breadcrumbs/src/Breadcrumbs.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FC, useState } from 'react';
+import cn from 'classnames';
+
+import {
+  Breadcrumb,
+  BreadcrumbLinkProps,
+  BreadcrumbsProps,
+} from './Breadcrumbs.types';
+
+import styles from './styles/default.module.css';
+
+const DefaultLink: React.FC<BreadcrumbLinkProps> = ({
+  link,
+  className,
+  children,
+  ...rest
+}) => (
+  <a className={className} href={link} {...rest}>
+    {children}
+  </a>
+);
+
+const Breadcrumbs: FC<BreadcrumbsProps> = ({
+  breadcrumbs,
+  dataTestId,
+  className,
+}) => {
+  if (breadcrumbs.length === 0) {
+    return null;
+  }
+
+  const Link = DefaultLink as (props: BreadcrumbLinkProps) => JSX.Element;
+
+  const firstBreadcrumb = breadcrumbs[0];
+  const lastBreadcrumb = breadcrumbs[breadcrumbs.length - 1];
+  const hiddenBreadcrumbs = breadcrumbs.slice(1, -1);
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks, @typescript-eslint/naming-convention
+  const [showHiddenBreadcrumbs, setShowHiddenBreadcrumbs] = useState(false);
+
+  const toggleHiddenBreadcrumbs = () => {
+    setShowHiddenBreadcrumbs(!showHiddenBreadcrumbs);
+  };
+
+  return (
+    <ul
+      className={cn(styles.breadcrumbWrapper, className)}
+      data-test-id={dataTestId}
+    >
+      {firstBreadcrumb?.link ? (
+        <li key={`${firstBreadcrumb.title}-${0}`}>
+          <Link link={firstBreadcrumb.link}>
+            <span itemProp="name">{firstBreadcrumb.title}</span>
+          </Link>
+        </li>
+      ) : (
+        firstBreadcrumb.title && (
+          <li key={firstBreadcrumb.title}>
+            <span itemProp="name">{firstBreadcrumb.title}</span>
+          </li>
+        )
+      )}
+      {window.innerWidth < 768 &&
+      hiddenBreadcrumbs.length > 0 &&
+      !showHiddenBreadcrumbs ? (
+        <li className={styles.threeDots} onClick={toggleHiddenBreadcrumbs}>
+          &#8230;
+        </li>
+      ) : (
+        <>
+          {hiddenBreadcrumbs &&
+            hiddenBreadcrumbs.map((breadcrumb, i) => {
+              if (breadcrumb.link && i + 1 !== breadcrumbs.length) {
+                return (
+                  // eslint-disable-next-line react/no-array-index-key
+                  <li key={`${breadcrumb.title}-${i}`}>
```

Comment:
Using the index as the key can cause some issues. You can use the `uuid` package to create a unique id for each item. Theres an example in the pagination component https://github.com/hey-car/heycar-uikit/blob/ea31197b42b48b39a31078e6a747fa61f4b63516/packages/pagination/src/Pagination.tsx#L37